cfme/tests/v2v/test_infrastructure_mapping.py

```diff
@@ -0,0 +1,49 @@
+"""Test to validate End-to-End migrations- functional testing."""
+import pytest
+
+from cfme.fixtures.provider import rhel7_minimal
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.markers.env_markers.provider import ONE_PER_VERSION
+
+pytestmark = [
+    pytest.mark.provider(
+        classes=[RHEVMProvider], selector=ONE_PER_VERSION, required_flags=["v2v"], scope="module"
+    ),
+    pytest.mark.provider(
+        classes=[VMwareProvider],
+        selector=ONE_PER_TYPE,
+        fixture_name="source_provider",
+        required_flags=["v2v"],
+        scope="module",
+    ),
+    pytest.mark.usefixtures("v2v_provider_setup", "add_host_creds", "assign_conversion_tags"),
```

Comment:
I think it would be good to structure the fixtures such that `add_host_creds` and `assign_conversion_tags` are included just by the fixture tree, and are not required to be specified directly in the test module or function.

Reply:
OSP does not required add_host_creds and assign_conversion_tags fixture , hence cannot combine these with v2v_provider fixture . They need to be separate so that only v2v_provider_Setup can be called in case of OSP

Reply:
Gotcha. So, if they are required by v2v_provider_setup, and they check the `provider` type - just yielding immediately when its OSP - then all V2V tests that use provider setup would get those actions by default. Thoughts?